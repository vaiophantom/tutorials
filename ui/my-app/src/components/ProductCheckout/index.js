import { currencyFormat } from "../../utils/helper";

const ProductCheckout = (props) => {
  const { data } = props;

  return (
    <div className="product-checkout-screen">
      <div style={{fontWeight:"bold"}}>{data.name}</div>
      <div>{currencyFormat(data.price * data.quantity)} $</div>
    </div>
  );
};
export default ProductCheckout;
