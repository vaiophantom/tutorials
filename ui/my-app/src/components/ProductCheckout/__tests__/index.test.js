import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import ProductCheckout from "../";
import CONSTANT from "../../../constants";

const mockProps = {
  data: CONSTANT.PRODUCT_CART,
};
it("renders ProductCheckout", async () => {
  const { container } = render(<ProductCheckout {...mockProps}/>, { wrapper: BrowserRouter });
  expect(container).toMatchSnapshot();
});
