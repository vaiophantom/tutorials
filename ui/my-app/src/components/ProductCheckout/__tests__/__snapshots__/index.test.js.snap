// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProductCheckout 1`] = `
<div>
  <div
    class="product-checkout-screen"
  >
    <div
      style="font-weight: bold;"
    >
      name
    </div>
    <div>
      100.00
       $
    </div>
  </div>
</div>
`;
